gateway/router.go

```diff
@@ -109,8 +109,8 @@ func (r *Router) handleReceive() {
 					msgIDs = append(msgIDs, gw.handleMessage(msg, br)...)
 				}
 				// only add the message ID if it doesn't already exists
-				if _, ok := gw.Messages.Get(msg.ID); !ok && msg.ID != "" {
-					gw.Messages.Add(msg.ID, msgIDs)
+				if _, ok := gw.Messages.Get(msg.Protocol + " " + msg.ID); !ok && msg.ID != "" {
+					gw.Messages.Add(msg.Protocol+" "+msg.ID, msgIDs)
```

Comment:
I'm seeing in testing that protocol is already reflected in msg.ID at this point, and so isn't needed. (This is the origin of the extra space as well, as `msg.Protocol` is empty)

Also, msg.Protocol isn't populated anymore, so this is an empty string. I think this got lost in a refactor, and we should bring it back in `populateReceivedMessage` even if it isn't helpful here :)